.github/workflows/test.yaml

```diff
@@ -0,0 +1,20 @@
+name: Go test
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+jobs:
+  test:
+    name: Go Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
+      - name: Run unit tests
+        run: go run mage.go test:go
```

Comment:
[question] should we add this script under `hack/mage` to use as a wrapper instead of executing this? However, I'm fine with this as well.